sg_jira/handlers/entity_issue_handler.py

```diff
@@ -987,18 +975,114 @@ def _get_shotgun_assignment_from_jira_issue_change(
                 # Note that we are dealing here with a Jira raw value dict, not
                 # a jira.resources.Resource instance.
                 jira_user = jira_issue["fields"]["assignee"]
-                sg_user = self._shotgun.find_one(
-                    "HumanUser",
-                    [["email", "is", jira_user["emailAddress"]]],
-                    ["email", "name"]
+                sg_user = self._jira_user_to_shotgun(
+                    shotgun_field, user_id=to_assignee, jira_user=jira_user
+                )
+                current_sg_assignment = sg_user
+        return current_sg_assignment
+
+    def _jira_server_user_to_shotgun(self, shotgun_field, user_id, jira_user=None, raise_on_missing_user=True):
+        """
+        Resolve the Shotgun user associated to the JIRA user passed in.
+
+        This method should be called against a JIRA local server.
+
+        :param str shotgun_field: Field to sync the value to in Shotgun.
+        :param str user_id: Value of the to or from of a JIRA changelog.
+        :param dict jira_user: Value of the user section of the webhook payload.
+        :param bool raise_on_missing_user: Indicate how to handle unknown users.
+
+        :returns: A Shotgun user entity dictionary.
+        :raises InvalidJiraValue: Raised if the user could not be found and ``raise_on_missing_user`` is True.
+        """
+        if jira_user is not None:
+            emailAddress = jira_user["emailAddress"]
+        elif user_id is not None:
+            emailAddress = self._jira.user(user_id, payload="key").emailAddress
+        else:
+            raise RuntimeError("jira_user or user_id cannot be both None.")
```

Comment:
"...cannot both be None."

Could we also add a little bit of info in the error message to describe why this might have happened? That's assuming there's a single or most likely cause that's explainable.